lib/include/hse_ikvdb/cndb.h

```diff
@@ -28,371 +30,135 @@ struct kvdb_rparams;
 
 /* MTF_MOCK_DECL(cndb) */
 
-/**
- * cndb_alloc() - allocate a new cndb
- * @ds:       dataset
- * @captgt:   (input/output)capacity target for the cndb
- * @oid1_out: new cndb MDC mlog id
- * @oid2_out: new cndb MDC mlog id
- *
- * If @captgt is not specified, CNDB_CAPTGT_DEFAULT is used.  If specified,
- * the argument is updated with the allocated capacity, which may be higher
- * than requested.
- */
+struct cndb;
+struct cndb_txn;
+
 /* MTF_MOCK */
 merr_t
-cndb_alloc(struct mpool *ds, u64 *captgt, u64 *oid1_out, u64 *oid2_out);
-
-/**
- * cndb_create() - initialize a new cndb with no cn KVSes.
- * @ds:       dataset
- * @captgt:   capacity target for the cndb
- * @oid1:     new cndb MDC mlog id
- * @oid2:     new cndb MDC mlog id
- *
- */
+cndb_create(struct mpool *mp, size_t size, uint64_t *oid1_out, uint64_t *oid2_out);
+
 /* MTF_MOCK */
 merr_t
-cndb_create(struct mpool *ds, u64 captgt, u64 oid1, u64 oid2);
-
-/**
- * cndb_drop() - Delete a cndb and its constituent cn KVSes.
- * @ds:   dataset
- * @oid1: cndb MDC mlog id #1
- * @oid2: cndb MDC mlog id #2
- *
- * This is currently a NO-OP
- */
-merr_t
-cndb_drop(struct mpool *ds, u64 oid1, u64 oid2);
-
-/**
- * cndb_open() - Open a cndb
- * @ds:   dataset
- * @read_only: readonly flag
- * @cndb_entries: max number of entries in a CNDB's in memory representation.
- * @oid1: mlog id #1 of the cndb MDC
- * @oid2: mlog id #2 of the cndb MDC
- * @health: reference to the container kvdb's health struct
- * @cndb_out: cndb handle
- *
- * Implementation notes:
- *   - Cndb MDC processing will occur as part of cndb_open, rolling MDC
- *     transactions forward or backward as appropriate.
- *   - Diagnostic tools such as cn_metrics, and cn_kbdump will use
- *     cn secret sauce APIs to interact with cn trees w/o fully
- *     activating them.
- */
+cndb_destroy(struct mpool *mp, uint64_t oid1, uint64_t oid2);
+
 /* MTF_MOCK */
 merr_t
 cndb_open(
-    struct mpool *      ds,
-    bool                read_only,
-    atomic_ulong       *ikvdb_seqno,
-    size_t              cndb_entries,
-    u64                 oid1,
-    u64                 oid2,
-    struct kvdb_health *health,
+    struct mpool        *mp,
+    u64                  oid1,
+    u64                  oid2,
+    bool                 rdonly,
     struct kvdb_rparams *rp,
-    struct cndb **      cndb_out);
+    struct kvdb_health  *health,
+    struct cndb        **cndb_out);
```

Comment:
Document how rdonly==true impacts behavior or API.  Internal to CNDB it means replay is skipped.  I guess that means CNDB might present an older view of the system than would exist if replay was not skipped?  With respect to compactions, node splits, etc that doesn't matter much.  But what about KVS create and destroy?

Reply:
Ok, CNDB doesn't do much with that flag right now. I'll fix this in a subsequent PR.